core/app/models/spree/promotion_handler/free_shipping.rb

```diff
@@ -23,10 +23,12 @@ def activate
       private
 
       def promotions
-        Spree::Promotion.active.where(
-          id: Spree::Promotion::Actions::FreeShipping.pluck(:promotion_id),
-          path: nil
-        )
+        Spree::Promotion.active.
+          joins(:promotion_codes).where({
+            id: Spree::Promotion::Actions::FreeShipping.pluck(:promotion_id),
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.